src/components/chat/comment-card.tsx

```diff
@@ -14,22 +15,27 @@ interface IState {
   commentAdded: boolean;
   commentText: string;
 }
+const minTextAreaHeight = 35;
 
 @inject("stores")
 @observer
 export class CommentCard extends BaseComponent<IProps, IState> {
+  state = {
+    commentTextAreaHeight: minTextAreaHeight,
+    commentAdded: false,
+    commentText: "",
+  };
 
-  constructor(props: IProps) {
-    super(props);
-    this.handleCommentTextAreaChange = this.handleCommentTextAreaChange.bind(this);
-    this.handleCancelPost = this.handleCancelPost.bind(this);
-    this.handleSendPost = this.handleSendPost.bind(this);
-
-    this.state = {
-      commentTextAreaHeight: 35,
-      commentAdded: false,
-      commentText: "",
-    };
+  escFunction = (event: any) => {
+    if(event.keyCode === 27) {
+      this.setState({commentTextAreaHeight: minTextAreaHeight, commentAdded: false, commentText: ""});
+    }
+  }
+  componentDidMount(){
+    document.addEventListener("keydown", this.escFunction, false);
+  }
+  componentWillUnmount(){
+    document.removeEventListener("keydown", this.escFunction, false);
   }
```

Comment:
You shouldn't need to use DOM event handlers, here -- `onKeyDown={handleEscKey}` is simpler and should do the same thing. Also, since you're adding these handlers to the document, there's the potential to intercept keystrokes intended for other controls which would also be avoided by using React event handling mechanisms.